packages/gestalt/src/Text.js

```diff
@@ -91,26 +83,37 @@ export default function Text({
     underline && typography.underline,
     weight === 'bold' && typography.fontWeightBold,
     weight === 'normal' && typography.fontWeightNormal,
-    truncate && typography.truncate,
+    // `lineClamp` overrides `truncate`
+    truncate && !isNotNullish(lineClamp) && typography.truncate,
```

Comment:
It's currently possible to pass both `truncate` and `lineClamp`, though unlikely since `lineClamp` is undocumented. Regardless, `lineClamp` explicitly overrides `truncate`.